python/ray/data/read_api.py

```diff
@@ -2372,7 +2372,7 @@ def from_spark(
 
 @PublicAPI
 def from_huggingface(
-    dataset: Union["datasets.Dataset", "datasets.IterableDataset"],
+    dataset: Union["datasets.Dataset", "datasets.IterableDataset"], parallelism=-1
```

Comment:
can we add `parallelism` in the docstring? let's make sure we state that the user should not need to set `parallelism` in most cases since it is auto-configured--similar to the docstring for `parallelism` under `read_parquet()` and other read methods.

in addition, we should also clarify that this is only used when the distributed parquet read is possible (i.e. when it is a public dataset with no transformations), otherwise it's a single node read.